config/minimalist-templates/button_card_templates.yaml

```diff
@@ -842,53 +842,107 @@ card_battery:
+            infix = "-charging-wireless";
+            break;
+          case "ac":
+            infix = "-charging";
+            break;
+          default:
+            infix = "";
+        }
+      }
+
+      if (battery_level == 100) {
         icon = "mdi:battery";
-      } else if (battery >= 90) {
-        icon = "mdi:battery-90";
-      } else if (battery >= 80) {
-        icon = "mdi:battery-80";
-      } else if (battery >= 70) {
-        icon = "mdi:battery-70";
-      } else if (battery >= 60) {
-        icon = "mdi:battery-60";
-      } else if (battery >= 50) {
-        icon = "mdi:battery-50";
-      } else if (battery >= 40) {
-        icon = "mdi:battery-40";
-      } else if (battery >= 30) {
-        icon = "mdi:battery-30";
-      } else if (battery >= 20) {
-        icon = "mdi:battery-20";
-      } else if (battery >= 10) {
-        icon = "mdi:battery-10";
+      } else if (battery_level >= 90) {
+        icon = "mdi:battery"+infix+"-90";
+      } else if (battery_level >= 80) {
+        icon = "mdi:battery"+infix+"-80";
+      } else if (battery_level >= 70) {
+        icon = "mdi:battery"+infix+"-70";
+      } else if (battery_level >= 60) {
+        icon = "mdi:battery"+infix+"-60";
+      } else if (battery_level >= 50) {
+        icon = "mdi:battery"+infix+"-50";
+      } else if (battery_level >= 40) {
+        icon = "mdi:battery"+infix+"-40";
+      } else if (battery_level >= 30) {
+        icon = "mdi:battery"+infix+"-30";
+      } else if (battery_level >= 20) {
+        icon = "mdi:battery"+infix+"-20";
+      } else if (battery_level >= 10) {
+        icon = "mdi:battery"+infix+"-10";
       } else {
-        icon = "mdi:battery-outline";
+        icon = "mdi:battery"+infix+"-outline";
       }
 
       return icon;
     ]]]
   label: |
     [[[
-      var battery = variables.ulm_card_battery_attribute !== null
+      var battery_level = variables.ulm_card_battery_attribute !== null
         ? states[entity.entity_id].attributes[variables.ulm_card_battery_attribute]
         : states[entity.entity_id].state;
-      return battery + "%";
+      return battery_level + "%";
     ]]]
   name: |
     [[[
       return states[entity.entity_id].attributes.friendly_name;
     ]]]
   styles:
     icon:
-      - color: 'rgba(var(--color-theme),0.9)'
+      - color: >
+          [[[
+              var color = 'rgba(var(--color-theme),0.9)';
+              var battery_level = variables.ulm_card_battery_attribute !== null
+              ? states[entity.entity_id].attributes[variables.ulm_card_battery_attribute]
+              : states[entity.entity_id].state;
+
+              // battery_level<battery_level_danger and battery_level_warning not set
+              color = variables.ulm_card_battery_battery_level_danger !== null 
+                      && variables.ulm_card_battery_battery_level_danger>=battery_level
+                      && variables.ulm_card_battery_battery_level_warning == null? 
```

Comment:
Why does it matter whether the warning level has been set here? 
In my opinion, if the battery level is lower than danger level we should always choose the danger level color.